sygnal/apnspushkin.py

```diff
@@ -278,17 +278,22 @@ async def dispatch_notification(self, n, device, context):
                             retry_delay, twisted_reactor=self.sygnal.reactor
                         )
 
-    def _get_payload_event_id_only(self, n):
+    def _get_payload_event_id_only(self, n, device):
         """
         Constructs a payload for a notification where we know only the event ID.
         Args:
             n: The notification to construct a payload for.
+            device (Device): Device information to which the constructed payload
+            will be sent.
 
         Returns:
             The APNs payload as a nested dicts.
         """
         payload = {}
 
+        if device.data is not None and device.data["default_payload"] is not None:
+            payload = device.data["default_payload"]
```

Comment:
I think this will fail if there is no `default_payload` in `device.data`.

Also, I think you need to make sure you copy the `default_payload` dict here, otherwise any updates you do to it in the next few lines will overwrite the data on the `Device`.

```suggestion
        if device.data:
            payload.update(device.data.get("default_payload", {}))
```

Reply:
> I think this will fail if there is no `default_payload` in `device.data`.

Actually it doesn't. There are tests handling this. But copying makes sense.

Reply:
I don't see any tests for this...

Reply:
There are some tests without a `data.default_payload`, which was enough for this check. Like: https://github.com/matrix-org/sygnal/blob/d63e70f163df7d4e068b3c8d1c07134b380a5701/tests/test_apns.py#L26

Reply:
that has no `data` either. the problem came when there is a `data` but no `default_payload`.

Reply:
Ah sorry, you're right 👍